components/Service-Management.tsx

```diff
@@ -0,0 +1,142 @@
+// components/service-management.tsx
+'use client'
+
+import React, { useState } from 'react';
+import { useQuery, useMutation } from 'convex/react';
+import { api } from '@/convex/_generated/api';
+import { Button } from '@/components/ui/button';
+import { Input } from '@/components/ui/input';
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { Dialog, DialogContent, DialogHeader, DialogTitle } from '@/components/ui/dialog';
+import { toast } from '@/components/ui/use-toast';
+import { Spinner } from '@/components/ui/spinner';
+import { Service } from '@/types/service';
+
+export const ServiceManagement: React.FC = () => {
+    const [isAddingService, setIsAddingService] = useState(false);
+    const [editingService, setEditingService] = useState<Service | null>(null);
+    const [newService, setNewService] = useState<Partial<Service>>({});
+
+    const services = useQuery(api.services.list);
+    const addService = useMutation(api.services.add);
+    const updateService = useMutation(api.services.update);
+    const deleteService = useMutation(api.services.remove);
+
+    const handleAddService = async () => {
+        try {
+            await addService(newService);
+            setIsAddingService(false);
+            setNewService({});
+            toast({ title: "Success", description: "Service added successfully" });
+        } catch (error) {
+            console.error("Error adding service:", error);
+            toast({ title: "Error", description: "Failed to add service", variant: "destructive" });
+        }
+    };
+
+    const handleUpdateService = async () => {
+        if (!editingService) return;
+        try {
+            await updateService({ id: editingService._id, ...editingService });

```

Comment:
**Suggestion:** To ensure data consistency and prevent potential runtime errors, consider adding checks to ensure that `editingService` and `newService` objects have all required properties before attempting to update or add services. [possible issue, importance: 8]
```suggestion
            if (validateService(editingService)) {
              await updateService({ id: editingService._id, ...editingService });
            }
            if (validateNewService(newService)) {
              await addService(newService);
            }
```